pkg/volume/local_volume/local_volume.go

```diff
@@ -251,10 +280,24 @@ func (m *localVolumeMounter) SetUpAt(dir string, fsGroup *int64) error {
 		return err
 	}
 
-	// TODO: how to prevent multiple mounts with conflicting fsGroup?
 	return volume.SetVolumeOwnership(m, fsGroup)
 }
 
+// isBindMounted checks whether the volume has beend mounted before.
+func (m *localVolumeMounter) isBindMounted() (bool, error) {
+	mountpoints, err := m.mounter.List()
+	if err != nil {
+		return false, err
+	}
+	for _, mountpoint := range mountpoints {
```

Comment:
Also check out the kubelet volume manager code under pkg/kubelet/volume.  You may be able to leverage some information under the actual state of the world.